server/controllers/article.js

```diff
@@ -346,6 +361,69 @@ class ArticleController {
       });
     }
   }
+
+  /**
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @return {json} res
+   * @description deletes an article object with given slug in param.
+  */
+  static reportOnArticle(req, res, next) {
+    const slug = req.params.article_slug;
+    const { userId, userName } = req;
+    const { reasonForReport, reportBody } = req.body;
+
+    const { error, isValid } = ReportInputValidation.validateArticleReportInput(req.body);
+    if (!isValid) {
+      return res.status(400).json({
+        error,
+        status: 'error'
+      });
+    }
+
+    Article.findOne({
+      where: { slug }
+    })
+      .then((article) => {
+        if (!article) {
+          return res.status(404).json({
+            status: 'error',
+            errors: {
+              message: 'Article does not exist',
+            }
+          });
+        }
+        return ReportsOnArticle.findOne({
+          where: { userId }
+        })
+          .then((found) => {
+            if (found) {
+              return res.status(400).json({
+                status: 'error',
+                error: {
+                  message: 'We already have your report on this article'
```

Comment:
A user should be able to report an article more than once. Line `398-410`

Reply:
Okay. I guess that makes more sense